src/main/scala/mesosphere/marathon/api/AuthResource.scala

```diff
@@ -43,14 +40,30 @@ trait AuthResource extends RestResource {
 
     val requestFacade = new RequestFacade(request)
     result(authenticator.authenticate(requestFacade)) match {
-      case Some(identity) if isAllowed(identity) => fn(identity)
-      case Some(identity)                        => response(authorizer.handleNotAuthorized(identity, requestFacade, _))
-      case None                                  => response(authenticator.handleNotAuthenticated(requestFacade, _))
+      case Some(identity) if isAuthorized(identity) => fn(identity)
+      case Some(identity) =>
+        response(authorizer.handleNotAuthorized(identity, requestFacade, _))
+      case None => response(authenticator.handleNotAuthenticated(requestFacade, _))
     }
   }
 
-  def isAllowedToView(pathId: PathId)(implicit identity: Identity): Boolean = {
-    authorizer.isAuthorized(identity, ViewAppOrGroup, pathId)
+  def doIfAuthorized[Resource](request: HttpServletRequest,
+                               identity: Identity,
```

Comment:
You can pass the identity as implicit parameter (Given that the executing block is wrapped in an `doIfAuthenticated`)